### DISCUSSION TITLE
Magic Link Login Not Working

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#7092] [Questions]
We have an app that works locally, but we've just pushed it to a hosted service and the magic link is not working. When I click it I see a "login" action in the audit trail but in the app:

`supabaseClient.auth.user();`

returns null.

I'm not sure how to start debugging this. Any help would be appreciated.

**Comment by GaryAustin1** [comment]
You will likely need to provide more info.  
Sounds like you get the magic link, then when you click on it your app starts and some audit trail shows login.  Then at some point auth.user() is null.
Auth.user returning null means at the instant you called it there is not a user object (session) built in the client.  This could be time sensitive as there are a couple of cases where it takes time to load the session, verify the token, etc.
You say it is running locally and then pushed it to hosted... assumption is you mean your app is moved to hosted.  Where is your Subabase instance at?
Has your hosted URL been adjusted in the Supabase server?  Does the link contain the correct URL and a token?
You might want to show more code around the login process...

**Reply by jaredcosulich** [reply]
Hi Gary,

This is the section of code in question. It gets called from a useEffect hook on the page when it loads:

```
  console.log("LISTEN")
  supabaseClient.auth.onAuthStateChange(async (event, session) => {
    console.log('event', event)
    console.log('session', session)
    console.log('session.user', session.user)
  });

  const authUser = supabaseClient.auth.user();

  console.log("AUTH USER", authUser);
```

authUser is null, the onAuthStateChange is never called. When I refresh the page the authUser is still null.

> You say it is running locally and then pushed it to hosted... assumption is you mean your app is moved to hosted. Where is your Subabase instance at?
> Has your hosted URL been adjusted in the Supabase server? Does the link contain the correct URL and a token?
> You might want to show more code around the login process...

We were running the code locally, but against the Supabase directly. We are now running the code on hosted servers while still pointing to Supabase directly. I believe the url has changed on the supabase servers. I've changed:

Auth -> Settings -> General -> Site URL (and Additional redirect URLs)

Is there some place else we need to change from localhost:3000?

**Reply by GaryAustin1** [reply]
Look at the redirect_to part of the magic link URL.  It should point to the page your code is going to run on.
If not then you probably need to add "redirectTo" option to the signIn call.  Just a quick test but it appears to just go to root without the redirect parameter, but not really digging into it.

When I click on the link in email on my test app on a net server, (and I have redirectTo set to go to the correct page) it does generate a SIGNED_IN event.
User is null on a call in main code, then SIGNED_IN event occurs and then user is valid:
![image](https://user-images.githubusercontent.com/54564956/171304818-bdf94484-5838-4701-8d41-5803d33ed9c5.png)

**Reply by jaredcosulich** [reply]
Ok, thank you for the help. I figured out the issue. We were running some of the logic through an NPM package which was still pointing to localhost (which was running and returning the null user because it was not authenticated with localhost).
